### ISSUE TITLE
QUsb2Snes hasn't worked since 7.11

### ISSUE THREAD

**Issue by codemann8** [Skarsnik/QUsb2snes#47]
I've tried every version after 7.11 up until the latest 7.16.1.  All of which crash every time I attempt to transfer a ROM to the console thru QFile2Snes, both thru the context menu and thru the actual app.  I am running on SD2Snes Pro and SuperNT.  I've made sure to close out of all programs that could have the potential of interfering, including firewall and AV.

I've been excited to utilize some of the new features, however, none of the updates seemed to have fixed this issue. So I've been stuck on 7.11 in the meantime.

**Comment by Skarsnik** [comment]
I would need like a debug log. go in the misc menu of qusb2snes and enable debug log and restart it.
I had some weird issue with usb2snes firmware lately, I will also look at what is changed since 0.7.10

**Comment by codemann8** [comment]
``
2020-03-30T12:44:24              default - Debug: Runing QUsb2Snes version  "0.7.16.1"
2020-03-30T12:44:24              default - Debug: Compiled against Qt 5.12.6 , running 5.12.6
2020-03-30T12:44:24             WSServer - Debug: Adding Device Factory  "SD2Snes"
2020-03-30T12:44:24                APPUI - Debug: Scanning  "C:/Programs/QUsb2Snes/apps/QFile2Snes"
2020-03-30T12:44:24                APPUI - Debug: Searching for  "C:/Programs/QUsb2Snes/apps/QFile2Snes/qusb2snesapp.json"
2020-03-30T12:44:24              default - Debug: "Found a json description file qusb2snesapp.json"
2020-03-30T12:44:24                APPUI - Debug: Found an application Name :  "QFile2Snes"  Description :  "A basic file explorer for the usb2snes firmware" Folder :  "C:/Programs/QUsb2Snes/apps/QFile2Snes" Icon :  "icone.png" Exe : "QFile2Snes" QtApp : true
2020-03-30T12:44:24                APPUI - Debug: Adding  "C:/Programs/QUsb2Snes/apps/QFile2Snes"  -  "QFile2Snes"
2020-03-30T12:44:24                APPUI - Debug: Name :  "QFile2Snes"  Description :  "A basic file explorer for the usb2snes firmware" Folder :  "C:/Programs/QUsb2Snes/apps/QFile2Snes" Icon :  "icone.png" Exe : "QFile2Snes" QtApp : true
2020-03-30T12:44:24             WSServer - Debug: WebSocket server started : listenning  QHostAddress(QHostAddress::Any) port :  8080
2020-03-30T12:44:41             WSServer - Debug: New connection from  ""
2020-03-30T12:44:41             WSServer - Debug: New connection accepted  "Websocket 3055d40" "" QHostAddress("::1")
2020-03-30T12:44:41             WSServer - Debug: "Websocket 3055d40" received  "{\n    \"Opcode\": \"DeviceList\",\n    \"Space\": \"SNES\"\n}\n"
2020-03-30T12:44:41             WSServer - Debug: Request is  USB2SnesWS::DeviceList
2020-03-30T12:44:41             WSServer - Debug: Executing request :  0 Created at QTime("12:44:41.442") - USB2SnesWS::DeviceList SD2Snes::SNES () () WSServer::RequestState::NEW for "Websocket 3055d40"
2020-03-30T12:44:41             WSServer - Debug: Device List
2020-03-30T12:44:41       SD2SNESFactory - Debug: "COM4" "USB-SERIAL CH340" "" Busy :  false
2020-03-30T12:44:41       SD2SNESFactory - Debug: "COM1" "Communications Port" "" Busy :  false
2020-03-30T12:44:41       SD2SNESFactory - Debug: "COM3" "sd2snes USB RS-232 Emulation" "DEMO00000000" Busy :  false
2020-03-30T12:44:41             WSServer - Debug: "Websocket 3055d40" >> "{\n    \"Results\": [\n        \"SD2SNES COM3\"\n    ]\n}\n"
2020-03-30T12:44:41             WSServer - Debug: Server request finished -  0 Created at QTime("12:44:41.442") - USB2SnesWS::DeviceList SD2Snes::SNES () () WSServer::RequestState::NEW processed in  94  ms
2020-03-30T12:44:41             WSServer - Debug: "Websocket 3055d40" received  "{\n    \"Opcode\": \"Attach\",\n    \"Operands\": [\n        \"SD2SNES COM3\"\n    ],\n    \"Space\": \"SNES\"\n}\n"
2020-03-30T12:44:41             WSServer - Debug: Request is  USB2SnesWS::Attach
2020-03-30T12:44:41             WSServer - Debug: Executing request :  1 Created at QTime("12:44:41.539") - USB2SnesWS::Attach SD2Snes::SNES () ("SD2SNES COM3") WSServer::RequestState::NEW for "Websocket 3055d40"
2020-03-30T12:44:41             WSServer - Debug: "SD2Snes"
2020-03-30T12:44:41             WSServer - Debug: Found device "SD2SNES COM3" from "SD2Snes" State :  ADevice::CLOSED
2020-03-30T12:44:41             WSServer - Debug: Attaching  "Websocket 3055d40"  to  "SD2SNES COM3"
2020-03-30T12:44:41             WSServer - Debug: Trying to open device
2020-03-30T12:44:41              SD2SNES - Debug: Error  QSerialPort::NoError "No error"
2020-03-30T12:44:41              SD2SNES - Debug: Opening Serial connection :  true
2020-03-30T12:44:41              SD2SNES - Debug: DTR changed :  true
2020-03-30T12:44:41              SD2SNES - Debug: BaudRate :  9600
2020-03-30T12:44:41              SD2SNES - Debug: Databits :  QSerialPort::Data8
2020-03-30T12:44:41              SD2SNES - Debug: DataTerminalReady :  true
2020-03-30T12:44:41              SD2SNES - Debug: Parity :  QSerialPort::NoParity
2020-03-30T12:44:41              SD2SNES - Debug: FlowControl :  QSerialPort::NoFlowControl
2020-03-30T12:44:41              SD2SNES - Debug: Stop bits :  QSerialPort::OneStop
2020-03-30T12:44:41             WSServer - Debug: Adding device "SD2SNES COM3"
2020-03-30T12:44:41             WSServer - Debug: Added device :  "SD2SNES COM3"
2020-03-30T12:44:41             WSServer - Debug: Server request finished -  1 Created at QTime("12:44:41.539") - USB2SnesWS::Attach SD2Snes::SNES () ("SD2SNES COM3") WSServer::RequestState::NEW processed in  19  ms
2020-03-30T12:44:41             WSServer - Debug: "Websocket 3055d40" received  "{\n    \"Opcode\": \"Name\",\n    \"Operands\": [\n        \"SendToDialog\"\n    ],\n    \"Space\": \"SNES\"\n}\n"
2020-03-30T12:44:41             WSServer - Debug: Request is  USB2SnesWS::Name
2020-03-30T12:44:41             WSServer - Debug: Executing request :  2 Created at QTime("12:44:41.558") - USB2SnesWS::Name SD2Snes::SNES () ("SendToDialog") WSServer::RequestState::NEW for "Websocket 3055d40"
2020-03-30T12:44:41             WSServer - Debug: Server request finished -  2 Created at QTime("12:44:41.558") - USB2SnesWS::Name SD2Snes::SNES () ("SendToDialog") WSServer::RequestState::NEW processed in  0  ms
2020-03-30T12:44:45             WSServer - Debug: "SendToDialog" received  "{\n    \"Opcode\": \"PutFile\",\n    \"Operands\": [\n        \"/ROMs/_alttpr/_msu1/ff7/alttpr - NoGlitches-open-ganon_0xvJEoQlMJ.sfc\",\n        \"200000\"\n    ],\n    \"Space\": \"SNES\"\n}\n"
2020-03-30T12:44:45             WSServer - Debug: Request is  USB2SnesWS::PutFile
2020-03-30T12:44:45             WSServer - Debug: Device is  ADevice::READY
2020-03-30T12:44:45             WSServer - Debug: Executing request :  3 Created at QTime("12:44:45.051") - USB2SnesWS::PutFile SD2Snes::SNES () ("/ROMs/_alttpr/_msu1/ff7/alttpr - NoGlitches-open-ganon_0xvJEoQlMJ.sfc", "200000") WSServer::RequestState::NEW for "SendToDialog"
2020-03-30T12:44:45              SD2SNES - Debug: CMD :  SD2Snes::PUT SD2Snes::FILE 0 "/ROMs/_alttpr/_msu1/ff7/alttpr - NoGlitches-open-ganon_0xvJEoQlMJ.sfc"
2020-03-30T12:44:45              SD2SNES - Debug: >> "5553424101000000" - 252-272 :  "002000002f524f4d732f5f616c747470722f5f6d"
2020-03-30T12:44:45              SD2SNES - Debug: Written :  512  bytes - Write count 0
2020-03-30T12:44:45             WSServer - Debug: Request executed
2020-03-30T12:44:45             WSServer - Debug: "SendToDialog" Received binary data 1024 "789c00429c0c429c0b429c40219c41219c42219c4321a9808d002118fbc228a900005b5c0088a1e23020018920c0872200ffa1eaa512f0fc588016a5f62920f003eed70fa5f62910d007add70f2901d009e61a201e84220080a120fc85641280d3006cc15f55462ea4c96cc96c1633007dd2f4904521f7819cd3e39e84c1ccb0d3d780818682a382a39898f8989999f29bed9b9df79d9ec2840c0ca40c0c0202020202020202020002020209020c02020902020e02a410b961808503b97d808504b999808505dc030078c2305c4680a18ba900005b4bab5c81e0a0eaad2b01d00dad4021cd3301d01a9c40218015cd3301f0108d40218d3301c9f2b0038d30019c2b01ad2d01d00dad4121cd3101d00e9c412180098d31018d41219c2d01ad2e018d4221ad2f018d43219c2e019c2f01a9808d00219c0c42a512d008e61220e08920d183ad2a01f0034c2d82a5968d2321a5978d2421a5988d2521a5998d3021a59a8d3121a59c8d3221a59d8d3221a59e8d3221a51c8d2c21a51d8d2d21a51e8d2e21a51f8d2f21ad20018d0d21ad21018d0d21ad24018d0e21ad25018d0e21ad1e018d0f21ad1f018d0f21ad22018d1021ad23018d1021a5e48d1121a5e58d1121a5ea8d1221a5eb8d1221a5958d0621a5948d05212907c907d0249c1c219c1c219c1d219c1d21ad38068d1f21ad39068d1f21ad3a068d2021ad3b068d2021ad2801f016ad1142a9388d09429c0a429c07429c0842a9a18d0042a5138d0021a59b8d0c42c230ab2b7afa6840204793a5ff8d09429c0a42a9a18d0042a5968d2321a5978d2421a5988d2521a5998d3021a59a8d3121a59c8d3221a59d8d3221a59e8d3221a51c8d2c21a51d8d2d21a51e8d2e21a51f8d2f21ad20018d0d21ad21018d0d21ad24018d0e21ad25018d0e21ad1e018d0f21ad1f018d0f21ad22018d1021ad23018d1021a5e48d1121a5e58d1121a5ea8d1221a5eb8d1221a5138d0021a59b8d0c42c2303baaad0a1f1b8e0a1fab2b7afa684078c23048da5a0b8b4babe230ad2a01d036ad11421029ad2801f0242c124250fbad30068d1121ad31068d11219c12219c1221ad280110089c2801a9818d0042c230ab2b7afa6840ad1142c2303baaad0a1f1b8e0a1fab2b7afa6840c220a988018502a980018016c220a97f008502a90003800ac220a97f008502a9ec0185009c15219c1621a908188d10439c1443a900008d1243a900208d1543a0028c0b42a2808e15219c16218d1543a908198d1043a901008d12438c0b42a50285009c1521a900608d1621a908188d1043a900008d1243a900088d15438c0b428e15218d1543a900608d1621a908198d1043a901008d12438c0b42e2206b9c16405c9285a400ad19428501a50085f2a845fa25f285f684faa50185f0a845f825f085f484f860ad1a428500ad1b" Pending size:  ()
2020-03-30T12:44:45             WSServer - Debug: true 2046
2020-03-30T12:44:45              SD2SNES - Debug: Written :  1024  bytes - Write count 1
2020-03-30T12:44:45             WSServer - Debug: "SendToDialog" Received binary data 1024 "428501a50085f3a845fb25f385f784fba50185f1a845f925f185f584f960a260a9f09d01089d05089d09089d0d089d11089d15089d19089d1d089d81089d85089d89089d8d089d91089d95089d99089d9d089d01099d05099d09099d0d099d11099d15099d19099d1d099d81099d85099d89099d8d099d91099d95099d99099d9d098a38e920aa109760000000000005000a000f80a4c0a400a540a5009020906090e091a090c090009140910093409380938094c094e094c095009520954095809440968096a0968097c096e0960097809400984098a09880948094809480948094c09a009b809480948094809480948094c09b009c409c809cc09c009d409d8094409f809fc09fe09f00a080948094809400a1809480948094809480948094809480948094809480948094809480948094c0980099c099e099009a209a409a609a80948094809480948094809480948094809a80948094809480948094809480948094809480948094809480948094809480948094809480948094809480948094809480948094809480948094809480948094809480948094809480948094809480948094809480948094809480948094e000e00060008000c001e00040000000800000004000000000000000000000000e000400060010000600080000002000400000002000400000004000800000004000800040b300b4c0b4a01c980a0aaabd230a0a0a1d220a0a0a1d210a0a0a1d200a99000abd270a0a0a1d260a0a0a1d250a0a0a1d240a99010abd2b0a0a0a1d2a0a0a0a1d290a0a0a1d280a99020abd2f0a0a0a1d2e0a0a0a1d2d0a0a0a1d2c0a99030a8888888810a2c231ae0001bd96938dcc0a6900028dce0abdf4958dd00a186900028dd20aae0201bd52988dd40aae0401bd52988dd60ae210ae0701bd9c848dc00a186980018dc20aae0801bdac848dc40a1869c0008dc60aad090129f8004a4aa8ad09010aaabdb2848dc80a18bb7db2858dca0aadc3022903000aaabd94848dd80a186900018dda0aaf0dc07e3a8f0dc07ed02fa90900a68ce0b5f004e0bcd003a917008f0dc07eaf0fc07e18690004c9000cd003a900008f0fc07e186980a68ddc0aaf13c07e3a8f13c07ed02aaf15c07eaae8e8e00cd002a2008a8f15c07ebdd2858f13c07ea980b2187dde858de00a186960008de20aade80a0a6940b98dec0a6900028dee0aadea0a0a6940b98df00a6900028df20aadf40a0a6940b58df60a6900028df80ae2206084037a8400c23029ff000aa8688501c8b7008500e230a403dc000084057a8402c23029ff0085030a6503a8688503c8b7028500c8b7028501e230a405dc0000c230acfe01a2fe03a900009d00009d00049d00089d000c9d00109d00149d00189d001ccacad0e48f00c57e80321f70afe10370c9aa55f007a900008fe10370af" Pending size:  ()
2020-03-30T12:44:45             WSServer - Debug: true 1022
2020-03-30T12:44:45              SD2SNES - Debug: Written :  1024  bytes - Write count 2
2020-03-30T12:44:45             WSServer - Debug: Data for pending request 0 0
2020-03-30T12:44:45             WSServer - Debug: "SendToDialog" Putting data in queue
2020-03-30T12:44:45             WSServer - Debug: "SendToDialog" Putting data in queue
``

**Comment by codemann8** [comment]
[log-debug.txt](https://github.com/Skarsnik/QUsb2snes/files/4404521/log-debug.txt)

**Comment by codemann8** [comment]
[log.txt](https://github.com/Skarsnik/QUsb2snes/files/4775958/log.txt)
[log-debug.txt](https://github.com/Skarsnik/QUsb2snes/files/4775959/log-debug.txt)
 @Skarsnik This still continues to happen.  The problem I'm seeing is when I try to navigate 3 folders deep in QFile2Snes. I can enter the first directory, but when trying to go thru the next directory, a blank shows, No "." or "..", just full blank.  This still continues to be a hard stop, still running on 0.7.11.

**Comment by Skarsnik** [comment]
Hm, interessting, it seems like the file listing is buggy. Not sure why it broke since 0.7.11

**Comment by codemann8** [comment]
@Skarsnik I finally figured out the issue. There must've been some junk data that got added in a previous version of QFile, likely during a botched write operation. And there must've been something added on 7.12 and after like reading additional file metadata or something extra that didn't occur in previous versions. Because it seemed like just this one file folder caused the program to freeze and crash.  After deleting a couple of the files in that folder, it then started working. So some file was borked, and some new operation was attempting to run against that file. I'm happy to say I'm on 7.17.2 after several months of banging my head.
